SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/KYCView.kt

```diff
@@ -0,0 +1,582 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.Dimension
+import androidx.lifecycle.viewmodel.compose.viewModel
+import app.cybrid.sdkandroid.BuildConfig
+import app.cybrid.sdkandroid.R
+import app.cybrid.sdkandroid.components.kyc.view.IdentityVerificationViewModel
+import app.cybrid.sdkandroid.core.Constants
+import app.cybrid.sdkandroid.ui.Theme.robotoFont
+import com.withpersona.sdk2.inquiry.Inquiry
+import com.withpersona.sdk2.inquiry.InquiryResponse
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class KYCView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0):
+    Component(context, attrs, defStyle) {
+
+    enum class KYCViewState { LOADING, REQUIRED, VERIFIED, ERROR, REVIEWING }
+
+    private var currentState = mutableStateOf(KYCViewState.LOADING)
+    var identityViewModel: IdentityVerificationViewModel? = null
+
+    init {
+
+        LayoutInflater.from(context).inflate(R.layout.kyc_component, this, true)
+        this.composeView = findViewById(R.id.composeContent)
+    }
+
+    fun setViewModel(identityViewModel: IdentityVerificationViewModel) {
+
+        this.identityViewModel = identityViewModel
+        this.identityViewModel?.UIState = this.currentState
+        this.initComposeView()
+
+        if (BuildConfig.DEBUG) {
+            GlobalScope.launch { identityViewModel.createCustomerTest() }
+        } else {
+            identityViewModel.getCustomerStatus()
+        }
+    }
+
+    private fun initComposeView() {
+
+        this.composeView?.let { compose ->
+            compose.setContent {
+                KYCView(
+                    viewModel = this.identityViewModel!!,
+                    currentState = this.currentState
+                )
+            }
+        }
+    }
+
+    companion object {
+
+        fun openPersona(
+            identityViewModel: IdentityVerificationViewModel,
+            getInquiryResult: ManagedActivityResultLauncher<Inquiry, InquiryResponse>
+        ) {
+
+            val id = identityViewModel.latestIdentityVerification?.identityVerificationDetails?.personaInquiryId ?: ""
+            val inquiry = Inquiry.fromInquiry(id)
+                .build()
+            getInquiryResult.launch(inquiry)
+        }
+    }
+}
+
+/**
+ * Composable Functions for Accounts
+ * **/
+
+@Composable
+fun KYCView(
```

Comment:
This file is very long. Are we able to break it up into multiple files, one per `@Composable` view?

Reply:
Solved!